src/core_plugins/kibana/public/context/app.js

```diff
@@ -0,0 +1,100 @@
+import _ from 'lodash';
+
+import uiModules from 'ui/modules';
+import contextAppTemplate from './app.html';
+import './components/loading_button';
+import './components/size_picker/size_picker';
+import {
+  createInitialQueryParametersState,
+  QueryParameterActionsProvider,
+  QUERY_PARAMETER_KEYS,
+} from './query_parameters';
+import { QueryActionsProvider } from './query';
+
+const module = uiModules.get('apps/context', [
+  'elasticsearch',
+  'kibana',
+  'kibana/config',
+  'kibana/notify',
+  'ngRoute',
+]);
+
+module.directive('contextApp', function ContextApp() {
+  return {
+    bindToController: true,
+    controller: ContextAppController,
+    controllerAs: 'contextApp',
+    restrict: 'E',
+    scope: {
+      anchorUid: '=',
+      columns: '=',
+      indexPattern: '=',
+      predecessorCount: '=',
+      successorCount: '=',
+      sort: '=',
+      discoverUrl: '=',
+    },
+    template: contextAppTemplate,
+  };
+});
+
+function ContextAppController($scope, Private) {
+  const queryParameterActions = Private(QueryParameterActionsProvider);
+  const queryActions = Private(QueryActionsProvider);
+
+  this.state = createInitialState(this.discoverUrl);
+
+  this.actions = _.mapValues(Object.assign(
+    {},
+    queryParameterActions,
+    queryActions,
+  ), (action) => (...args) => action(this.state)(...args));
+
+  $scope.$watchGroup([
+    () => this.state.rows.predecessors,
+    () => this.state.rows.anchor,
+    () => this.state.rows.successors,
+  ], _.spread(this.actions.setAllRows));
```

Comment:
I can not follow the reasoning behind that. Do we really assume that a developer working on Kibana is incapable of intuitively understanding what `spread` does?

Regarding `$watchGroup`, I've heard two contradicting philosophies by now:
* Don't fight the framework, instead embrace what it has to offer.
* Don't use the framework features that do exactly what you need, because developers shouldn't have to know the semantics of those features.

We can't have it both ways.